### DISCUSSION TITLE
Show an "unread" count for notifications?

### DISCUSSION THREAD

**Discussion by nikclayton** [pachli/pachli-android#17] [Ideas]
Users have proposed showing an (optional) count of unread notifications on the notification tab icon.

I think this is difficult, because Mastodon doesn't have a concept of a notification being read or unread. A notification exists, and it either continues to exist, or the user deletes it, and that's it.

Pachli could maintain its own local read/unread state for each notification. However, that could lead to more user confusion -- the user could read a notification in another app, or in the web client, or in Pachli on two different devices, and see the same notification marked as "unread", and conclude that Pachli is at fault.

Mastodon does have a notification "marker" -- this records the ID of the most recent notification you have seen. This would allow the user to do something like "Mark everything below this notification as unread", but it doesn't let you, say, have 5 notifications and mark notifications 1, 2, and 5 as read, keeping 3 and 4 as unread.

Suggestions for how to present this to the user in an understandable way are welcomed.

**Comment by nikclayton** [comment]
I have just filed https://github.com/mastodon/mastodon/issues/26866 as a feature request against Mastodon, which would allow this to be implemented.

**Comment by Russssty** [comment]
I've been using the app for a while now, and I have a suggestion.  Please let me know what you think.

Since issue #2686 remains open and the Pachli (2.9.2) uses status bar notifications, let's link the status bar notification count to a red badge on the Notifications tab icon.

The solution is not perfect (no sync between clients), but still an improvement.

SCENARIO A

Precondition:
- the app is closed (backgrounded)
- there are 3 notification in Android status bar:
  - X favorited your post
  - X replied to your post
  - Y favorited your post

1. Open the app 
App UI: Home timeline is visible
Android status bar: 3 Pachli notifications
Notifications tab icon: 3

2.  Navigate to Notification tab
App UI: Notification tab is visible, new notifications fetched
Android status bar: no notification
Notifications tab icon: no count
   

SCENARIO B

Precondition:
- the app is open
- Home timeline visible
- no new notification in Android status bar

1. New notification received
App UI: Home timeline is visible
Android status bar: 1 Pachli notifications
Notifications tab icon: 1

2.  Navigate to Notification tab
App UI: Notification tab is visible, new notification fetched
Android status bar: no notification
Notifications tab icon: no count


SCENARIO C

Precondition:
- the app is open
- Notification tab visible
- no new notification in Android status bar

1. New notification received
App UI: Notification tab visible
Android status bar: 1 Pachli notifications
Notifications tab icon: 1

2.  Scroll to fetch the notification
App UI: Notification tab is visible, new notification fetched
Android status bar: no notification
Notifications tab icon: no count
